pkg/account/component.go

```diff
@@ -68,3 +78,95 @@ func (c *component) UpdatePassword(ctx context.Context, currentPassword, newPass
 
 	return updateError
 }
+
+func (c *component) GetAccount(ctx context.Context) (api.AccountRepresentation, error) {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+	var realm = ctx.Value(cs.CtContextRealm).(string)
+
+	var userRep api.AccountRepresentation
+	userKc, err := c.keycloakClient.GetAccount(accessToken, realm)
+
+	if err != nil {
+		c.logger.Warn("err", err.Error())
+		return userRep, err
+	}
+
+	userRep = api.ConvertToAPIAccount(userKc)
+
+	return userRep, nil
+}
+
+func (c *component) UpdateAccount(ctx context.Context, user api.AccountRepresentation) error {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+	var realm = ctx.Value(cs.CtContextRealm).(string)
+	var userID = ctx.Value(cs.CtContextUserID).(string)
+	var username = ctx.Value(cs.CtContextUsername).(string)
+	var userRep kc.UserRepresentation
+
+	// get the "old" user representation
+	oldUserKc, err := c.keycloakClient.GetAccount(accessToken, realm)
+	if err != nil {
+		c.logger.Warn("err", err.Error())
+		return err
+	}
+
+	var emailVerified, phoneNumberVerified *bool
+
+	// when the email changes, set the EmailVerified to false
+	if user.Email != nil && oldUserKc.Email != nil && *oldUserKc.Email != *user.Email {
```

Comment:
Copied/pasted from management/component.go ? Couldn't it be factorized ?

Reply:
Not exactly a copy paste. It diverged as it use an AccountRepresentation which has no attributes. So I don't see a clean way to factorize it.